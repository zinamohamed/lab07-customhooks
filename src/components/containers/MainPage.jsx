import React from 'react';
import CustomHook  from '../hooks/ArnoldHook';
import CharacterList from '../presentations/CharacterList';

const MainPage = () => {

  const { loading, characters } = CustomHook();
 
  if(loading) return <h1>Loading...</h1>;
  return <CharacterList characters={characters}/>;
   
};

export default MainPage; 