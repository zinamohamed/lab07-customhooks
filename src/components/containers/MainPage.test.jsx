/** 
* @jest-environment jsdom
*/
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import  MainPage  from './MainPage';

describe('Testing MainPage Container', () => { 

    it('should render a list of hey arnold characters', async () => { 
    render(<MemoryRouter> <MainPage/> </MemoryRouter>);

    screen.getByText('Loading...');

    const ul = await screen.findByRole('list', { name: 'characters' });

    return waitFor(() => { 
      expect(ul).not.toBeEmptyDOMElement();
      expect(ul).toMatchSnapshot();

    });


  });
});