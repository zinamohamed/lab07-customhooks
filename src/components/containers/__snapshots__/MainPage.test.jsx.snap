// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MainPage Container should render a list of hey arnold characters 1`] = `
<ul
  aria-label="characters"
>
  <li>
    <div>
      <p>
        Character: 
        Arnold Shortman
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f51"
      >
        <img
          alt="Arnold Shortman"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Grandpa Phil
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f52"
      >
        <img
          alt="Grandpa Phil"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Grandma Gertie
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f53"
      >
        <img
          alt="Grandma Gertie"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/7/7b/DB87AC2A-C2C2-4DAB-AECF-B9F7273C746B.png/revision/latest/scale-to-width-down/306?cb=20190816101257"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Mitzi
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f54"
      >
        <img
          alt="Mitzi"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Arnie
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f55"
      >
        <img
          alt="Arnie"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Miles Shortman
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f56"
      >
        <img
          alt="Miles Shortman"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Stella Shortman
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f57"
      >
        <img
          alt="Stella Shortman"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/f/fa/Stella.jpg/revision/latest?cb=20101225143956"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Grandpa's father
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f58"
      >
        <img
          alt="Grandpa's father"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Grandpa's grandfather
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f59"
      >
        <img
          alt="Grandpa's grandfather"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Oskar Kokoshka
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f5a"
      >
        <img
          alt="Oskar Kokoshka"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/c/c5/Oskar_Kokoshka.png/revision/latest/scale-to-width-down/310?cb=20180707032418"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Suzie Kokoshka
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f5b"
      >
        <img
          alt="Suzie Kokoshka"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/f/fe/Suzie.jpg/revision/latest/scale-to-width-down/310?cb=20110223003447"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Nancy
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f5c"
      >
        <img
          alt="Nancy"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Ernie Potts
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f5d"
      >
        <img
          alt="Ernie Potts"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/6/67/Ernie.jpeg/revision/latest/scale-to-width-down/310?cb=20181213181427"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Mr. Hyunh
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f5e"
      >
        <img
          alt="Mr. Hyunh"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/6/60/Mr._Hyunh.png/revision/latest/scale-to-width-down/310?cb=20180707031441"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Mai Hyunh
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f5f"
      >
        <img
          alt="Mai Hyunh"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/8/83/Mai_Hyunh.jpeg/revision/latest/scale-to-width-down/310?cb=20190118192457"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Mr. Smith
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f60"
      >
        <img
          alt="Mr. Smith"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/8/80/Mr._Smith.jpg/revision/latest?cb=20110206233402"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Lana Vail
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f61"
      >
        <img
          alt="Lana Vail"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/762C69BE-23BA-48BA-9B14-6AF3246FCD3F.jpeg/revision/latest/scale-to-width-down/304?cb=20190106211607"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Mr. Purdy (character)
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f62"
      >
        <img
          alt="Mr. Purdy (character)"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/9/9c/Mr._Purdy_%28character%29.jpeg/revision/latest?cb=20190125214742"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Helga G. Pataki
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f63"
      >
        <img
          alt="Helga G. Pataki"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e5/Helga_Pataki.png/revision/latest/scale-to-width-down/266?cb=20190816021807"
        />
      </a>
    </div>
  </li>
  <li>
    <div>
      <p>
        Character: 
        Robert "Big Bob" Pataki
      </p>
      <a
        exact="true"
        href="/details/5da237699734fdcb7bef8f64"
      >
        <img
          alt="Robert \\"Big Bob\\" Pataki"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Big_Bob.jpg/revision/latest/scale-to-width-down/310?cb=20170428021615"
        />
      </a>
    </div>
  </li>
</ul>
`;
